ext/rbs_extension/parser.c

```diff
@@ -1023,8 +1031,13 @@ static VALUE parse_simple(parserstate *state) {
     return parse_symbol(state);
   }
   case tUIDENT: {
-    ID name = INTERN_TOKEN(state, state->current_token);
```

Comment:
This `INTERN_TOKEN` macro was calling `rb_intern3()`, which was slowly leaking memory by creating static ("immortal") symbols.

The replacement `rbs_constant_pool_find()` function uses `rb_check_id_cstr()` instead, which checks for a match without intern a new symbol if there wasn't one.